src/sentry/api/endpoints/user_authenticator_details.py

```diff
@@ -134,6 +134,14 @@ def delete(self, request, user, auth_id, interface_device_id=None):
             return Response(status=status.HTTP_204_NO_CONTENT)
 
         with transaction.atomic():
+            # if the user's organization requires 2fa,
+            # don't delete the last auth method
+            enrolled_methods = Authenticator.objects.all_interfaces_for_user(user, ignore_backup=True)
+            last_2fa_method = len(enrolled_methods) == 1
+            require_2fa = user.get_orgs_require_2fa().exists()
+
+            if require_2fa and last_2fa_method:
+                return Response(status=status.HTTP_500_INTERNAL_SERVER_ERROR)
```

Comment:
I think we should return something more useful than just a 500 server error. We should return back some 403 with a message indicating why.

https://httpstatuses.com/403